tests/warp/warp_test.go

```diff
@@ -315,54 +346,66 @@ func (w *warpTest) sendMessageFromSendingSubnet() {
 		Value:     common.Big0,
 		Data:      packedInput,
 	})
+
 	signedTx, err := types.SignTx(tx, w.sendingSubnetSigner, w.sendingSubnetFundedKey)
 	require.NoError(err)
+
 	log.Info("Sending sendWarpMessage transaction", "txHash", signedTx.Hash())
 	require.NoError(client.SendTransaction(ctx, signedTx))
-
 	log.Info("Waiting for new block confirmation")
 	<-newHeads
 	receiptCtx, cancel := context.WithTimeout(ctx, 10*time.Second)
 	defer cancel()
+
 	blockHash, blockNumber := w.getBlockHashAndNumberFromTxReceipt(receiptCtx, client, signedTx)
+	log.Info("Warp message included in block", "blockHash", blockHash, "blockNumber", blockNumber)
 
-	log.Info("Constructing warp block hash unsigned message", "blockHash", blockHash)
-	w.blockID = ids.ID(blockHash) // Set blockID to construct a warp message containing a block hash payload later
-	require.NoError(err)
+	return blockHash, blockNumber
+}
 
-	log.Info("Fetching relevant warp logs from the newly produced block")
-	logs, err := client.FilterLogs(ctx, ethereum.FilterQuery{
-		BlockHash: &blockHash,
-		Addresses: []common.Address{warp.Module.Address},
-	})
+// verifyAndExtractWarpMessage filters for the SendWarpMessage event using the
+// generated binding and sets w.addressedCallUnsignedMessage.
+func (w *warpTest) verifyAndExtractWarpMessage(
+	ctx context.Context,
+	client ethclient.Client,
+	blockNumber uint64,
+	sender common.Address,
+) {
+	require := require.New(ginkgo.GinkgoT())
+
+	log.Info("Filtering SendWarpMessage events using binding")
+	warpFilterer, err := warpbindings.NewIWarpMessengerFilterer(warp.Module.Address, client)
 	require.NoError(err)
-	require.Len(logs, 1)
 
-	// Check for relevant warp log from subscription and ensure that it matches
-	// the log extracted from the last block.
-	txLog := logs[0]
-	log.Info("Parsing logData as unsigned warp message")
-	unsignedMsg, err := warp.UnpackSendWarpEventDataToMessage(txLog.Data)
+	iter, err := warpFilterer.FilterSendWarpMessage(
+		&bind.FilterOpts{
+			Start:   blockNumber,
+			End:     &blockNumber,
+			Context: ctx,
+		},
+		[]common.Address{sender},
+		nil, // messageID filter: any
+	)
 	require.NoError(err)
+	defer iter.Close()
 
-	// Set local variables for the duration of the test
-	w.addressedCallUnsignedMessage = unsignedMsg
-	log.Info("Parsed unsignedWarpMsg", "unsignedWarpMessageID", w.addressedCallUnsignedMessage.ID(), "unsignedWarpMessage", w.addressedCallUnsignedMessage)
+	// Verify we got exactly one event with the correct data
+	require.True(iter.Next(), "expected at least one SendWarpMessage event")
+	event := iter.Event // event is *IWarpMessengerSendWarpMessage
```

Comment:
I don't think this comment is necessary since your IDE says this
```suggestion
	event := iter.Event
```